docs/specification.md

```diff
@@ -0,0 +1,1264 @@
+
+- Contract must not be paused.
+
+**burn**
+
+Types
+```
+tokenId = nat
+
+burnParam
+  ( address :sender
+  , tokenId :tokenId
+  , nat     :value
+  )
+
+burn = list burnParam
+```
+
+Parameter (in Michelson):
+```
+(list %burn
+  (pair
+    (address %sender)
+    (pair
+      (nat %tokenId)
+      (nat %value)
+  ))
+)
+```
+
+- Decreases balances for senders and their token types and the total supply of tokens by the given amount.
+
+- Each burning operation must happen atomically, so if one of them fails, 
+  then the whole operation must fail.
+
+- The operation must follow permission policies described above.
+
+- Sender must be minter and must have a sufficient amount of
+  funds to be destroyed and also be in glboal minter list.
+
+- A minter with 0 minting allowance is allowed to burn tokens.
+
+- Contract must not be paused.
+
+**Whitelisting (TBD)**
+----------------
+
+Whitelisting unctions for the TZUSDC token implementation which are outside the FA1.2 Tezos Token Standard.
+
+**addToWhitelist** address
+
+- Add an address to whitelist, enabling it to transfer and
+  approve tokens as well as modify allowances.
+
+- Sender must be a blacklister.
+
+**addToWhitelistBatch** (list address)
+
+- Adds each address to whitelist, enabling it to transfer and
+  approve tokens as well as modify allowances.
+
+- Sender must be a blacklister.
+
+**removeFromWhitelist** address
+
+- Remove address from whitelist.
+
+- Sender must be a blacklister.
+
+**removeFromWhitelistBatch** (list address)
+
+- Remove provided addresses from whitelist.
+
+- If some of specified addresses are not in whitelist, then
+  it fails returning list of those addresses.
+
+- Sender must be a blacklister.
+
+**checkWhitelisted** (void address bool)
+
+- Return a boolean value describing whether the
+  given address is present in whitelist.
+
+Role reassigning functions
+==========================
+
+**Owner**
+---------
+
+**transferOwnership**
+
+Types
+```
+transferOwnership =
+  ( address :owner
+  , address :receiver
+  )
+```
+
+Parameter (in Michelson):
+```
+(pair %transferOwnership
+  (address %owner)
+  (address %receiver)
+)
+```
+
+- Set token owner to a new address.
+
+- Sender must be current token owner.
+
+- The current owner retains his priveleges up until
+  `acceptOwnership` is called.
+
+- Can be called multiple times, each call replaces pending
+  owner with the new one. Note, that if proposed
+  owner is the same as the current one, then the call
+  is simply invalidated.
+
+**getOwner**
+
+Types
+```
+getOwner = (contract address :callback)
+```
+
+Parameter (in Michelson):
+```
+(pair %getOwner
+  (contract %callback address)
+)
+```
+
+- Returns current token owner address.
+
+**acceptOwnership**
+
+Types
+```
+acceptOwnership = unit
+```
+
+Parameter (in Michelson):
+```
+(unit %acceptOwnership)
+```
+
+- Accept ownership privileges.
+
+- Sender must be a pending owner.
+
+**Admin**
```

Comment:
Could you add "admin" to the list of roles and describe its capabilities?

Reply:
Actually we don't need it since due tot CENTRE specification it can only upgrade the contract which we agreed on not to for the time being